src/cascadia/TerminalSettingsModel/JsonUtils.h

```diff
@@ -696,8 +707,41 @@ namespace Microsoft::Terminal::Settings::Model::JsonUtils
     template<typename T, typename Converter>
     void SetValueForKey(Json::Value& json, std::string_view key, const T& target, Converter&& conv)
     {
-        // demand guarantees that it will return a value or throw an exception
-        *json.demand(&*key.cbegin(), (&*key.cbegin()) + key.size()) = conv.ToJson(target);
+        std::optional<Json::Value> out;
+        if constexpr (Detail::DeduceOptional<T>::IsOptional)
+        {
+            // FOR OPTION TYPES
+            // - if we have a value, set the value
+            // - otherwise, ignore it and don't write it out
+            if constexpr (Detail::DeduceOptional<T>::IsString)

```

Comment:
i find this deeply objectionable.

Reply:
(Leaving this comment for tracking) #8175 allowed for the removal of this hack.